src/js/tech/tech.js

```diff
@@ -339,13 +395,13 @@ class Tech extends Component {
    */
 
   textTracks() {
-    this.player_.textTracks_ = this.player_.textTracks_ || new TextTrackList();
-    return this.player_.textTracks_;
+    this.textTracks_ = this.textTracks_ || new TextTrackList();
```

Comment:
@gkatsev You are right, it is causing me a lot of trouble with the unit tests. I think using an event that the player listent to would be more appropriate though. Would it work?